lib/internal/streams/end-of-stream.js

```diff
@@ -147,17 +147,25 @@ function eos(stream, opts, callback) {
   if (opts.error !== false) stream.on('error', onerror);
   stream.on('close', onclose);
 
-  const closed = (wState && wState.closed) || (rState && rState.closed) ||
-    (wState && wState.errorEmitted) || (rState && rState.errorEmitted) ||
-    (wState && wState.finished) || (rState && rState.endEmitted) ||
-    (rState && stream.req && stream.aborted);
+  const closed = (
```

Comment:
Can we simplify this a bit? Perhaps at the very least group together similar dependencies, like:

```js
const closed = (
  (wState && (wState.closed || wState.errorEmitted)) ||
  (rState && (rState.closed || rState.errorEmitted || (stream.req && stream.aborted))) ||
  (
    (!writable || (wState && wState.finished)) &&
    (!readable || (rState && rState.endEmitted))
  )
);
```

Reply:
They are grouped?

Reply:
Unless there is a performance benefit to the above suggestion, I prefer @ronag's grouping because it's easier to read. Though it has more lines, it has less parentheses and the lines are ordered by the properties (e.g. `closed`) of the states. It reads like "is either state closed, or either state errorEmitted, or ..".

Reply:
I have not personally benchmarked it, so I cannot say if it is measurable or not. However, it _is_ reducing the number of duplicate checks so V8 _should_ be performing less work.

However, my code suggestion was merely _one_ possibility. I'm not opposed to rearranging the checks in other ways, such as introducing separate `if` statements, etc.

Reply:
I think prefer the current formatting and believe any performance implication here would be negligible in practice. This is not a hot path as far as I'm aware. A future simplification could be to use the `?.` operator.